shaderexp/main.zig

```diff
@@ -75,21 +80,24 @@ pub fn init(app: *App, core: *mach.Core) !void {
     bgl.release();
 }
 
-pub fn deinit(app: *App, core: *mach.Core) void {
+pub fn deinit(app: *App) void {
     app.fragment_shader_file.close();
-    core.allocator.free(app.fragment_shader_code);
+    allocator.free(app.fragment_shader_code);
 
     app.uniform_buffer.release();
     app.bind_group.release();
+
+    app.core.deinit();
```

Comment:
I think we should standardize on having this be a `defer app.core.deinit()` in all examples